pandas/tseries/offsets.py

```diff
@@ -1160,6 +1162,196 @@ def onOffset(self, dt):
     _prefix = 'MS'
 
 
+class SemiMonthOffset(SingleConstructorOffset):
+    _adjust_dst = True
+    _default_day_of_month = 15
+    _min_day_of_month = 2
+
+    def __init__(self, n=1, normalize=False, **kwds):
+        self.day_of_month = kwds.get('day_of_month',
+                                     self._default_day_of_month)
+
+        if not self._min_day_of_month <= self.day_of_month <= 27:
+            raise ValueError('day_of_month must be '
+                             '{}<=day_of_month<=27, got {}'.format(
+                                 self._min_day_of_month, self.day_of_month))
+        self.n = int(n)
+        self.normalize = normalize
+        self.kwds = kwds
+
+    @classmethod
+    def _from_name(cls, suffix=None):
+        if not suffix:
+            day_of_month = 15
+        else:
+            day_of_month = int(suffix)
+        return cls(day_of_month=day_of_month)
+
+    @property
+    def rule_code(self):
+        suffix = '-%s' % self.day_of_month
+        return self._prefix + suffix
+
+    @apply_wraps
+    def apply(self, other):
+        n = self.n
+        if not self.onOffset(other):
+            _, days_in_month = tslib.monthrange(other.year, other.month)
+            if 1 < other.day < self.day_of_month:
+                other += relativedelta(day=self.day_of_month)
+                if n > 0:
+                    # rollforward to 15 so subtract 1
+                    n -= 1
+            elif self.day_of_month < other.day < days_in_month:
+                other += relativedelta(day=self.day_of_month)
+                if n < 0:
+                    # rollforward to 15 in the negative direction so add 1
+                    n += 1
+                elif n == 0:
+                    n = 1
+
+        return self._apply(n, other)
+
+    def _apply(self, n, other):
+        """Handle specific apply logic for child classes"""
+        raise AbstractMethodError(self)
+
+    @apply_index_wraps
+    def apply_index(self, i):
+        # determine how many days away from the 1st of the month we are
+        days_from_start = i.to_perioddelta('M').asi8
+        delta = Timedelta(days=self.day_of_month - 1).asm8.astype(int)
```

Comment:
I'm guessing this fails on windows (will try to cast to int32), use `Timedelta(...).value`